cmd/launcher/launcher.go

```diff
@@ -166,7 +169,24 @@ func runLauncher(ctx context.Context, cancel func(), opts *launcher.Options) err
 
 	fcOpts := []flags.Option{flags.WithCmdLineOpts(opts)}
 	flagController := flags.NewFlagController(logger, stores[storage.AgentFlagsStore], fcOpts...)
-	k := knapsack.New(stores, flagController, db)
+	slogger := ctxlog.FromContextWithSlogger(ctx)
+	k := knapsack.New(stores, flagController, db, slogger)
+
+	slogLevel := slog.LevelInfo
+	sysLogMatcher := multislogger.SystemLogMatcher
+	if k.Debug() {
+		// if were in debug mode, we want to log everything including debug messages
+		// to the system log (stderr). When not in debug mode, were only logging logs
+		// with attribute system_log=true to the system log (stderr).
```

Comment:
I think this will work, but I wonder if it's simpler to have something like:
```go
	k.AddReplaceSlogHandler("system", ...)
	
	// If we're in debug mode, then we assume we want to echo _all_ logs to stderr. 
        if k.Debug() {
  	    k.AddReplaceSlogHandler("debugcli", slog.NewJSONHandler(os.Stderr, &slog.HandlerOptions{...}))
        }
```

Reply:
This makes the split very clean, and I think handles the windows case more easily.

Reply:
yea, that's cleaner, done